app/gui/src/dashboard/services/RemoteBackend.ts

```diff
@@ -1471,6 +1471,36 @@ export default class RemoteBackend extends Backend {
     }
   }
 
+  /** Download the project to a temporary location. */
+  async downloadProject(id: backend.ProjectId): Promise<DirectoryId> {
+    const details = await this.getProjectDetails(id, true)
+
+    invariant(details.url != null, 'The download URL of the project must be present.')
+
+    const queryString = new URLSearchParams({
+      downloadUrl: details.url,
+    })
+
+    const response = await this.client.get(`./api/cloud/download-project?${queryString}`)
+    const path = await response.text()
+
+    invariant(response.ok, 'The download-project response must have status OK.')
+    invariant(path.length !== 0, 'The download-project response body must be present.')
+
+    return DirectoryId(`directory-${path}` as const)
+  }
+
+  /** Upload the project. */
+  async uploadProject(id: backend.ProjectId, directoryId: backend.DirectoryId): Promise<void> {
+    const uploadPath = remoteBackendPaths.getProjectUploadPath(id)
+    const queryString = new URLSearchParams({
+      uploadUrl: `${$config.API_URL}/${uploadPath}`,
+      directory: extractIdFromDirectoryId(directoryId),
+    })
+
+    await this.client.get(`./api/cloud/upload-project?${queryString}`)
```

Comment:
I think it's better to have `/` instead of `./` here. Also we prefer url constructor functions instead of bare bone strings. Please take a look at similar methods and do it similarly

Reply:
It looks like this module uses strings. I wasn't able to find any examples of URL usages

Reply:
It was inspired by https://github.com/enso-org/enso/blob/28a1b039d746ac31b4f7ffc1b8493fb624421497/app/gui/src/dashboard/layouts/AssetContextMenu.tsx#L302-L304

Reply:
yeah fwiw:
- i prefer `./` so that it works when proxied with a base path (e.g. through code-server or openvscode-server)
  - however i guess that's not super robust - instead we could use `APP_BASE_URL`: https://github.com/enso-org/enso/blob/e002d942a5d49ec2d275fdeb8f0c3e62ebaa2906/app/gui/src/dashboard/utilities/appBaseUrl.ts
- dunno about url constructors here. *maybe* it's possible for it to work but i think this is more understandable. after all, only the path needs changing so idk if there are many advantages to using `new URL` instead (if there were query string changes involved i would absolutely recommend `new URL` though)